content/docs/building-apps/add-stellar-assets.mdx

```diff
@@ -0,0 +1,185 @@
+
+  * check the memo field to determine which user should receive the payment, and
+  * debit the user’s account with the amount of XLM they received.
+
+This is the role of `depositIntoPool`, described above. You pass this function as the `onmessage` option when you [stream payments](https://developers.stellar.org/docs/tutorials/follow-received-payments/):
+
+<CodeExample>
+
+```js
+
+  const stream = server.payments()
+    .forAccount(custodialAcc)
+    .join("transactions")
+    .stream({
+      onmessage: (payment) => {
+        const from = payment.source_account;
+        const customerId = payment.memo;
+        const amount = payment.amount;
+
+        let asset = sdk.Asset.native();
+        if (payment.asset_issuer) {
+          asset = new sdk.Asset(payment.asset_code, payment.asset_issuer);
+        }
+
+        depositIntoPool(from, customerId, asset, amount);
+      }
+    });
+```
+
+</CodeExample>
+
+When someone **outside** of your customer base wants to send lumens to one of your customers, instruct them to make an XLM payment to your pooled account address with the customer ID in the memo field of the transaction. Assuming you have set the memo required flag on your account (see [above](#setting-the-memo-required-flag-on-your-account)), [well-behaved](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md) wallets should enforce it and prevent users from forgetting to attach a memo to an incoming payment. However, to be on the safe side, you should make it incredibly clear to senders that their payment will end up in limbo if they fail to attach a valid one. You can learn more about the transaction memo field [here](https://developers.stellar.org/docs/glossary/transactions/#memo).
+
+When someone **inside** of your customer base wants to send lumens to *another* one of your customers, you have two choices: you can send a memo'd payment exactly as above--this lets you maintain an audit trail, ensures the balance exists in the custodial account, etc.--or you can do the exchange "off-chain", i.e. by exclusively adjusting balances within your database.
+
+### Submitting outgoing payments to the Stellar network
+When one of your customers wants to make an outgoing lumen payment, you must generate a Stellar transaction to send XLM. See [building transactions](https://stellar.github.io/js-stellar-sdk/TransactionBuilder.html) or the [payments tutorial](https://developers.stellar.org/docs/tutorials/send-and-receive-payments/#send-a-payment) for more information.
+
+The aforementioned function `withdrawFromPool` will prepare the corresponding operation whenever a withdrawal is requested. You can use these to queue up transactions (for periodic or batched submission) or submit them immediately. It's up to you how to architect this portion, we adopt the (simpler) latter approach here:
+
+<CodeExample>
+
+```js
+function onPooledPayment(customerId, dest, amount, asset) {
```

Comment:
This particular function is supposed to be much less "this is how you do it" and more "this is here for completeness sake". Refer to the quote above:

> You can use these to queue up transactions (for periodic or batched submission) or submit them immediately. It's up to you how to architect this portion, we adopt the (simpler) latter approach here:

Since we want to be less authoritative and more demonstrative with this particular function (e.g. putting `customerId` in the transaction memo is purely optional), I thought leaving off the details would hammer this point home. However, I think you're right that it's still useful & instructional to include them for clarity.

Reply:
Done in e317b984c21f37be0c0ddf12b8bc5720a7d436e9